File: LeetCode/2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        pos = [num for num in nums if num > 0]
        neg = [num for num in nums if num < 0]
        sol = []

        for i in range(len(pos)):
            sol.append(pos[i])
            sol.append(neg[i])
        
        return sol
        

File: LeetCode/2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        queue<int> pos, neg;
        vector<int> sol;

        for (int num : nums) {
            if (num > 0) pos.push(num);
            else neg.push(num);
        } //for

        while (!pos.empty()) {
            sol.push_back(pos.front());
            sol.push_back(neg.front());
            pos.pop();
            neg.pop();
        } //while

        return sol;
    } //rearr
};